File: app/src/main/java/com/mailerdaemon/app/impContacts/DiffUtilCallBack.java
package com.mailerdaemon.app.impContacts;

import androidx.annotation.Nullable;
import androidx.recyclerview.widget.DiffUtil;

import java.util.List;

public class DiffUtilCallBack extends DiffUtil.Callback {
    private final List<Contact> mOldEmployeeList;
    private final List<Contact> mNewEmployeeList;

    public DiffUtilCallBack(List<Contact> oldEmployeeList, List<Contact> newEmployeeList) {
        this.mOldEmployeeList = oldEmployeeList;
        this.mNewEmployeeList = newEmployeeList;
    }

    @Override
    public int getOldListSize() {
        return mOldEmployeeList.size();
    }

    @Override
    public int getNewListSize() {
        return mNewEmployeeList.size();
    }

    @Override
    public boolean areItemsTheSame(int oldItemPosition, int newItemPosition) {
        return mOldEmployeeList.get(oldItemPosition).getName().equals( mNewEmployeeList.get(
                newItemPosition).getName());
    }

    @Override
    public boolean areContentsTheSame(int oldItemPosition, int newItemPosition) {
        final Contact oldEmployee = mOldEmployeeList.get(oldItemPosition);
        final Contact newEmployee = mNewEmployeeList.get(newItemPosition);

        return oldEmployee.getName().equals(newEmployee.getName());
    }

    @Nullable
    @Override
    public Object getChangePayload(int oldItemPosition, int newItemPosition) {
        // Implement method if you're going to use ItemAnimator
        return super.getChangePayload(oldItemPosition, newItemPosition);
    }
}

File: app/src/main/java/com/mailerdaemon/app/impContacts/DiffUtilCallBack.kt
package com.mailerdaemon.app.impContacts

import androidx.recyclerview.widget.DiffUtil

class DiffUtilCallBack(
    private val mOldEmployeeList: List<Contact>,
    private val mNewEmployeeList: List<Contact>
) : DiffUtil.Callback() {

    override fun getOldListSize() = mOldEmployeeList.size

    override fun getNewListSize() = mNewEmployeeList.size

    override fun areItemsTheSame(oldItemPosition: Int, newItemPosition: Int) =
        (mOldEmployeeList[oldItemPosition].name == mNewEmployeeList[newItemPosition].name)

    override fun areContentsTheSame(oldItemPosition: Int, newItemPosition: Int): Boolean {
        return mOldEmployeeList[oldItemPosition] == mNewEmployeeList[newItemPosition]
    }

    override fun getChangePayload(oldItemPosition: Int, newItemPosition: Int) =
        super.getChangePayload(oldItemPosition, newItemPosition)
}
